src/Symfony/Component/Form/Tests/Extension/Core/Type/TextTypeTest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Tests\Extension\Core\Type;
+
+use Symfony\Component\Form\Test\TypeTestCase as TestCase;
+
+class TextTypeTest extends TestCase
+{
+    public function testSubmitNullReturnsNull()
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\TextType', 'name');
+
+        $form->submit(null);
+
+        $this->assertNull($form->getData());
+    }
+
+    public function testSubmitNullReturnsEmptyStringWithEmptyDataAsString()
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\TextType', 'name', array(
+            'empty_data' => '',
+        ));
+
+        $form->submit(null);
```

Comment:
@backbone87 Yes, the reasoning is to not break data transformers by passing `null` instead of an empty string.

Concerning `TextType` this PR does not introduce a "new" way, it just consider the setting of `empty_data` which was ignored before.

This is a bug fix, I'm not sure it should go in master though.

ping @webmozart